cogstat/cogstat_chart.py

```diff
@@ -725,6 +725,107 @@ def create_variable_pair_chart(data, meas_lev, x, y, result=None, raw_data=False
         graph = plt.gcf()
     return graph
 
+def create_multiple_variable_chart(data, meas_lev, x, type="raw"):
+    """Draw a chart relating more than two variables displaying raw data
+
+    Parameters
+    ----------
+    data : pandas dataframe
+    meas_lev : {'int', 'ord', 'nom', 'unk'}
+        Measurement level of the variables
+    x : str
+        Name of the x variable.
+    type : {'raw', 'collinearity'}
+        Displays raw data when 'raw', displays scatterplot matrix of explanatory variables when 'collinearity'.
+
+    Returns
+    -------
+    matplotlib chart
+        A matrix plot of the variables optionally containing the raw data.
+    """
+
+    if meas_lev == "int":
+        if type == "collinearity":
+            data = data[x]
+        if type in "raw":
+            fig, ax = plt.subplots(len(data.columns), len(data.columns), tight_layout=True)
+            fig.suptitle(_plt("Scatterplot matrix of variables"))
+            for i in range(0, len(data.columns)):
+                ax[i, 0].set_ylabel(data.columns[i])
+                ax[len(data.columns) - 1, i].set_xlabel(data.columns[i])
+                for j in range(0, len(data.columns)):
+                    if i == j:
+                        ax[i, j].hist(data.iloc[:, i])
+                    else:
+                        ax[i, j].scatter(data.iloc[:, i], data.iloc[:, j])
+        elif type is "collinearity":
+            if len(x) == 2:
+                ncols = 1
+            elif len(x) >= 2:
+                ncols = 2
+            import math
+            nrows = math.ceil(len(x)/2)
+            fig = plt.figure(tight_layout=True)
+            fig.suptitle(_plt("Scatterplot matrix of explanatory variables"))
+            x_done = []
+            ind = 1
+            for index_1, x_i in enumerate(x):
+                for index_2, x_j in enumerate(x):
+                    if x_i != x_j and [x_i, x_j] not in x_done and [x_j, x_i] not in x_done:
+                        ax = plt.subplot(nrows, ncols, ind)
+                        ax.scatter(data[x_i], data[x_j])
+                        ax.set_xlabel(x_i)
+                        ax.set_ylabel(x_j)
+                        x_done.append([x_i, x_j])
+                        ind += 1
+
+        graph = plt.gcf()
+        return graph
+    else:
+        return None
+
+def part_regress_plots(y, x, data):
+    """Draw a matrix of partial regression plots.
+
+    Parameters
+    ----------
+    x : str
+        Name of the x variable.
+    y : str
+        Name of the y variable.
+    data : pandas dataframe
+        The dataframe analysed.
+
+    Returns
+    -------
+    matplotlib chart
+        A matrix plot of the variables containing the residuals from the respective partial regressions.
+    """
+
+    if len(x) == 2:
```

Comment:
I'm not sure what this function does. Seemingly, x is not a string. Please, revise the docstring, comments, and variable names

Reply:
Docstring extended and comments added. I hope it's better explained now.